### PR TITLE
Issue 59: Replace Django with Fastapi

### PR DESCRIPTION
**Pull request by FreakyNobleGas**
This PR completely replaces Django with FastAPI in our API, while keeping functionality largely the same as before.

Changes:
- Replaced Django with FastAPI
- Added SQLModel and Alembic for connecting and managing our database
- Updated our database models. The biggest change here is for IGDB, we are using a single JSON column instead of multiple fields
- Updated Dockerfile to work with FastAPI
- Updated parse_datasets.py to create mock data

Migration Guide:
I found it to be much easier to simply recreate the game-recommender database. Django created a lot of tables and other metadata that doesn't do well with running migrations using Alembic.

I was able to drop the database this way:
```bash
# Log into postgres
psql -U USER game-recommender
```
```sql
DROP DATABASE game-recommender
```

Once you recreate the database, I updated the readme to include running migrations and the server itself. The flow should be very similar to what we had before.

Testing
-------
Here is a screenshot after populating the database using parse_dataset.py and hitting our GET /api/games endpoint.
image

### CODE REVIEW

File: src/api/Dockerfile
Comment (FreakyNobleGas) [inline src/api/Dockerfile:1]: I was thinking we should update the app dockerfile to be tagged with a similar naming scheme. (e.g. app-latest)

File: src/api/alembic.ini
Comment (FreakyNobleGas) [inline src/api/alembic.ini:1]: This is standard Alembic config stuff. I didn't make too many modifications here.

File: src/api/alembic/env.py
Comment (FreakyNobleGas) [inline src/api/alembic/env.py:1]: This file defines how alembic creates the migration files. This is also very close to the standard configuration that is automatically provided.

File: src/api/alembic/versions/aef0008118ec_base_revision.py
Comment (FreakyNobleGas) [inline src/api/alembic/versions/aef0008118ec_base_revision.py:1]: This is the base revision to create the Game, Platform, Multiplayer, and IGDBGame tables in PostgreSQL.

File: src/api/database.py
Comment (FreakyNobleGas) [inline src/api/database.py:1]: This class facilitates creating a session which is a connection to the database.

File: src/api/main.py
Comment (FreakyNobleGas) [inline src/api/main.py:1]: This where all the action happens. We define the fastapi server and the GET endpoint. In follow up PRs, I will add in the IGDB functionality.

File: src/api/models.py
Comment (FreakyNobleGas) [inline src/api/models.py:1]: This is where we define all of our DB tables. It's handled through SQLModel which is an abstraction on top of SQLAlchemy and Pydantic. SQLAlchemy is a popular ORM tool in Python and Pydantic facilitates data validation.

**Review by Wolfinbarger — APPROVED** [review]

**Pull request merged**